src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -628,3 +754,33 @@ def _check_shift_offsets_are_literals(node: itir.StencilClosure):
             if not all(isinstance(arg, (itir.Literal, itir.OffsetLiteral)) for arg in shift.args):
                 return False
         return True
+
+    @staticmethod
+    def _replace_ssa_identifiers(closure: itir.StencilClosure):
```

Comment:
This solves the problem but I don't think this is the right approach.

We have two separate issues here:

### 1. The SSA pass generates invalid identifiers

See the documentation of the SSA pass:
> Mutates the python AST, variable names will not be valid python names anymore.

Instead of adding invalid identifiers and then later removing them, we could simplify the process by not adding them in the first place, so my opinion is that we should change the SSA pass rather than changing all backends. This could be as simple as replacing the special characters the SSA pass uses with an underscore right after the main SSA pass in a post-pass.

### 2. Python identifiers may not be valid in backends

This has nothing to do with special characters added by the SSA pass, but we it may happen that user-written code has valid identifiers in Python, but not in a backend. We should come up with a scheme that maps these invalid Python identifiers to a more common character set and reuse that pass across all backends. This is unrelated to this PR.

Reply:
@tehrengruber For issue 1, is it possible to paremeterize `_UNIQUE_NAME_SEPARATOR` in the SSA pass? or can we just change `ᐞ` to an underscore?

Reply:
Just using an underscore is not an option as it will lead to collisions with other symbols. We need a way to generate collision free symbols. If we are able to do that for arbitrary symbol names and to a common character set as @petiaccja mentions both issues are solved.

As a side-note the documentation is wrong the identifiers are valid python names. I would need to look up the PEP, but `ᐞ` is fine.

Reply:
There are indeed two different questions here as @petiaccja said: the method used by the SSA pass to rename identifiers and the missing sanitation of the IR identifiers in case the source language of some code-generation backends doesn't support the same characters as Python. Previously I also thought there were an associated third question which is how DaCe deals internally with non-valid identifiers from inputs, but if I understood correctly what @edopao said, DaCe checks beforehand that the identifiers are valid, which means that for us the issue with the DaCe backend is exactly the same as with other backends with limited support for unicode symbols like  C++.

Regarding the first question, the current SSA pass generates of course valid Python identifiers because the current separator was explicit chosen from the set of valid python identifiers (check https://docs.python.org/3/reference/lexical_analysis.html#identifiers) as a readable character for humans when inspecting and debugging the ITIR code. If the current implementation is problematic it could be changed to a different one, but since this wouldn't remove the need for a pass to sanitize the identifiers before passing them to some backends, unless there is a new proposal which doesn't harm human readability, I don't think it would make the code better.

For the second question, I do think there is need to add some identifiers sanitation pass to deal with the non-supported characters in the backend language. I don't have a concrete proposal but it doesn't look like something hard to add. One tool could help is the `slugify` method from `boltons` (https://boltons.readthedocs.io/en/latest/strutils.html#boltons.strutils.slugify) directly exposed in `eve.utils` (IIRC).

Reply:
I realized that this pass is not immediately needed. It is possible to enable `lift_mode=FORCE_TEMPORARIES` on gt4py unit tests without this pass. I can introduce it in a separate PR, if needed, as a generic ITIR pass to sanitize symbols.